Lib/functools.py

```diff
@@ -992,21 +989,15 @@ def __get__(self, instance, owner=None):
                 f"instance to cache {self.attrname!r} property."
             )
             raise TypeError(msg) from None
-        val = cache.get(self.attrname, _NOT_FOUND)
-        if val is _NOT_FOUND:
-            with self.lock:
-                # check if another thread filled cache while we awaited lock
-                val = cache.get(self.attrname, _NOT_FOUND)
-                if val is _NOT_FOUND:
-                    val = self.func(instance)
-                    try:
-                        cache[self.attrname] = val
-                    except TypeError:
-                        msg = (
-                            f"The '__dict__' attribute on {type(instance).__name__!r} instance "
-                            f"does not support item assignment for caching {self.attrname!r} property."
-                        )
-                        raise TypeError(msg) from None
+        val = self.func(instance)
```

Comment:
The evaluation is now always executed, the cache is not used. 
I would have expected there to be a test for this. @carljm

Reply:
@eendebakpt The cache is used implicitly because we are inside the `__get__` method of a descriptor here, which will never be called in the first place if there's a value of the same name in the instance's `__dict__` (assuming it's a "non-data" descriptor, i.e. one with only a `__get__` method.) That's the basic idea behind `cached_property` -- once you set a value in the instance dict, the descriptor becomes irrelevant and it's just a normal attribute lookup from that point onward.

Reply:
Ah I see. Thanks!